### PR TITLE
Enable using wolfSSL cryptographic API engine in OpenVPN

### PR DESCRIPTION
**Pull request by AuthorReflex**
Maintainer: me / @mkrkn 

Compile tested: ramips/mt7620 TP-Link Archer C50 v1, ramips/mt7621 Xiaomi Mi router 3 Pro
Run tested: ramips/mt7620 TP-Link Archer C50 v1, ramips/mt7621 Xiaomi Mi router 3 Pro

Description: Support for wolfSSL has been upstreamed to the master OpenVPN branch in f6dca235ae560597a0763f0c98fcc9130b80ccf4 so we can use wolfSSL directly in OpenVPN. So no more needed differnt SSL engine for OpenVPN in systems based on wolfSSL library

### CODE REVIEW

**Comment by mkrkn** [comment]
Thanks for the PR.
TL;DR: First, add "(experimental)" to the package description of openvpn-wolfssl. Second, fix the branch the PR is based on, there should only be one commit in this PR.

I've noticed the introduction of wolfSSL support in OpenVPN and thought it would be useful given that OpenWrt now relies on wolfSSL for WPA3 support anyway.

However, it isn't ready for a release yet. With the disclaimers in the OpenVPN repo, I read them as caring less about wolfSSL than LibreSSL. The latter is already a second-class citizen in OpenVPN compared to OpenSSL / mbed TLS. My plan was to add the package when OpenVPN ships a stable release with wolfSSL support. After thinking it through I think it's OK to have openvpn-wolfssl in master, with a disclaimer about it being experimental and to test with openvpn-openssl if any issues arise.

Also, your branch is messed up, please rebase or do something else. This change should be contained in one commit, your PR currently contains three -- one of which is a merge commit, which shouldn't occur in the PR itself.

**Comment by AuthorReflex** [comment]
> 
> 
> Thanks for the PR.
> TL;DR: First, add "(experimental)" to the package description of openvpn-wolfssl. Second, fix the branch the PR is based on, there should only be one commit in this PR.
> 
> I've noticed the introduction of wolfSSL support in OpenVPN and thought it would be useful given that OpenWrt now relies on wolfSSL for WPA3 support anyway.
> 
> However, it isn't ready for a release yet. With the disclaimers in the OpenVPN repo, I read them as caring less about wolfSSL than LibreSSL. The latter is already a second-class citizen in OpenVPN compared to OpenSSL / mbed TLS. My plan was to add the package when OpenVPN ships a stable release with wolfSSL support. After thinking it through I think it's OK to have openvpn-wolfssl in master, with a disclaimer about it being experimental and to test with openvpn-openssl if any issues arise.
> 
> Also, your branch is messed up, please rebase or do something else. This change should be contained in one commit, your PR currently contains three -- one of which is a merge commit, which shouldn't occur in the PR itself.

Thanks for your reply, fixed.

**Comment by PolynomialDivision** [comment]
Your commit message should be prefixed with the package name:

    openvpn: ...

**Comment by AuthorReflex** [comment]
> 
> 
> Your commit message should be prefixed with the package name:
> 
> ```
> openvpn: ...
> ```

Fixed

**Comment by PolynomialDivision** [comment]
@AuthorReflex the build fails. Please have a look at your patches. The patch number could also be wrong. `0xx` is for backporting and `1xx` is for "awaiting upstream merge". Since this is already merged upstream, I think it should be `1xx?`.
Why did you remove the commit message of the patch (https://github.com/OpenVPN/openvpn/commit/f6dca235ae560597a0763f0c98fcc9130b80ccf4)? I think it is a good style if you add a patch to also include the commit message (but just my opinion)

However, @aparcar or @neheb could you have a look at this. Is it okay to back port functionality until it is merged upstream?
@mkrkn seems fine with merging it with "(experimental)" string. However, I am unsure if we merge such PRs or at least I don't want to decide.

**Comment by AuthorReflex** [comment]
> 
> 
> @AuthorReflex the build fails. Please have a look at your patches. The patch number could also be wrong. `0xx` is for backporting and `1xx` is for "awaiting upstream merge". Since this is already merged upstream, I think it should be `0xx?`.
> Why did you remove the commit message of the patch ([OpenVPN/openvpn@f6dca23](https://github.com/OpenVPN/openvpn/commit/f6dca235ae560597a0763f0c98fcc9130b80ccf4))? I think it is a good style if you add a patch to also include the commit message (but just my opinion)
> 
> However, @aparcar or @neheb could you have a look at this. Is it okay to back port functionality until it becomes a release upstream?
> @mkrkn seems fine with merging it with "(experimental)" string. However, I am unsure if we merge such PRs or at least I don't want to decide.

Fixed:
 rename net/openvpn/patches/{120-add-wolfssl-support.patch => 002-add-wolfssl-support.patch}
 included original commit message in patch
 rebased offset nums in net/openvpn/patches/210-build_always_use_internal_lz4.patch 

For success build may need enable support for OpenVPN in wolfSSL, see https://github.com/openwrt/openwrt/pull/4148

**Comment by PolynomialDivision** [comment]
The buildlog says:
```
Now at patch 220-disable_des.patch
Patch 220-disable_des.patch is unchanged
mkdir -p ./patches
rm -f ./patches/* 2>/dev/null >/dev/null
'/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/001-reproducible-remove_DATE.patch' -> './patches/001-reproducible-remove_DATE.patch'
'/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/002-add-wolfssl-support.patch' -> './patches/002-add-wolfssl-support.patch'
'/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/100-mbedtls-disable-runtime-version-check.patch' -> './patches/100-mbedtls-disable-runtime-version-check.patch'
'/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/115-fix-mbedtls-without-renegotiation.patch' -> './patches/115-fix-mbedtls-without-renegotiation.patch'
'/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/210-build_always_use_internal_lz4.patch' -> './patches/210-build_always_use_internal_lz4.patch'
'/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/220-disable_des.patch' -> './patches/220-disable_des.patch'
make[2]: Leaving directory '/home/runner/work/packages/packages/net/openvpn'
time: package/feeds/packages_ci/openvpn/wolfssl/refresh#1.56#0.49#1.89
make[1]: Leaving directory '/home/build/openwrt'
Dirty patches detected, please refresh and review the diff
Error: Process completed with exit code 1.
```

**Comment by AuthorReflex** [comment]
> 
> 
> The buildlog says:
> 
> ```
> Now at patch 220-disable_des.patch
> Patch 220-disable_des.patch is unchanged
> mkdir -p ./patches
> rm -f ./patches/* 2>/dev/null >/dev/null
> '/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/001-reproducible-remove_DATE.patch' -> './patches/001-reproducible-remove_DATE.patch'
> '/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/002-add-wolfssl-support.patch' -> './patches/002-add-wolfssl-support.patch'
> '/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/100-mbedtls-disable-runtime-version-check.patch' -> './patches/100-mbedtls-disable-runtime-version-check.patch'
> '/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/115-fix-mbedtls-without-renegotiation.patch' -> './patches/115-fix-mbedtls-without-renegotiation.patch'
> '/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/210-build_always_use_internal_lz4.patch' -> './patches/210-build_always_use_internal_lz4.patch'
> '/home/build/openwrt/build_dir/target-mips_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/220-disable_des.patch' -> './patches/220-disable_des.patch'
> make[2]: Leaving directory '/home/runner/work/packages/packages/net/openvpn'
> time: package/feeds/packages_ci/openvpn/wolfssl/refresh#1.56#0.49#1.89
> make[1]: Leaving directory '/home/build/openwrt'
> Dirty patches detected, please refresh and review the diff
> Error: Process completed with exit code 1.
> ```

How I can check patches myself? My build logs don't show any problems

**Comment by PolynomialDivision** [comment]
```
make package/openvpn/clean && make package/openvpn/prepare QUILT=1 V=s
```
Look for the path where the source files are now located

```
quilt push -a
````

**Comment by AuthorReflex** [comment]
> 
> 
> ```
> make package/openvpn/clean && make package/openvpn/prepare QUILT=1 V=s
> ```
> 
> Look for the path where the source files are now located
> 
> ```
> quilt push -a
> ```

Executed succesful,  no errors or warnings

```
Executed succesful,  no warnings or errors

```
user@machine:/mnt/D2/openwrt/openwrt-trunk$ make package/openvpn/clean && make package/openvpn/prepare QUILT=1 V=s
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
 make[1] package/openvpn/clean
 make[2] -C feeds/packages/net/openvpn clean
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[1]: Entering directory '/mnt/D2/openwrt/openwrt-trunk'
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/feeds/packages/net/openvpn'
mkdir -p /mnt/D2/openwrt/openwrt-trunk/dl
SHELL= flock /mnt/D2/openwrt/openwrt-trunk/tmp/.openvpn-2.5.2.tar.xz.flock -c '        /mnt/D2/openwrt/openwrt-trunk/scripts/download.pl "/mnt/D2/openwrt/openwrt-trunk/dl" "openvpn-2.5.2.tar.xz" "b12743836901f365efaf82ab2493967e1b21c21eb43ce9a8da1002a17c9c1dc8" "" "https://build.openvpn.net/downloads/releases/" "https://swupdate.openvpn.net/community/releases/"    '
touch /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_mips32_musl/openvpn-wolfssl/openvpn-2.5.2/.prepared_check
. /mnt/D2/openwrt/openwrt-trunk/include/shell.sh; xzcat /mnt/D2/openwrt/openwrt-trunk/dl/openvpn-2.5.2.tar.xz | tar -C /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_mips32_musl/openvpn-wolfssl/openvpn-2.5.2/.. -xf -
[ ! -d ./src/ ] || cp -fpR ./src/. /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_mips32_musl/openvpn-wolfssl/openvpn-2.5.2
rm -rf /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_mips32_musl/openvpn-wolfssl/openvpn-2.5.2/patches; mkdir -p /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_mips32_musl/openvpn-wolfssl/openvpn-2.5.2/patches
touch /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_mips32_musl/openvpn-wolfssl/openvpn-2.5.2/.quilt_used
touch /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_mips32_musl/openvpn-wolfssl/openvpn-2.5.2/.prepared
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/feeds/packages/net/openvpn'
time: package/feeds/packages/openvpn/wolfssl/prepare#0.24#0.10#0.42
make[1]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk'
user@machine:/mnt/D2/openwrt/openwrt-trunk$ cd /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_mips32_musl/openvpn-wolfssl/openvpn-2.5.2/patches
user@machine:/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_mips32_musl/openvpn-wolfssl/openvpn-2.5.2/patches$ /mnt/D2/openwrt/openwrt-trunk/staging_dir/host/bin/quilt push -a
Applying patch ../patches/001-reproducible-remove_DATE.patch
patching file src/openvpn/options.c

Applying patch ../patches/002-add-wolfssl-support.patch
patching file configure.ac
patching file src/openvpn/syshead.h

Applying patch ../patches/100-mbedtls-disable-runtime-version-check.patch
patching file src/openvpn/ssl_mbedtls.c

Applying patch ../patches/115-fix-mbedtls-without-renegotiation.patch
patching file src/openvpn/ssl_mbedtls.c

Applying patch ../patches/210-build_always_use_internal_lz4.patch
patching file configure.ac

Applying patch ../patches/220-disable_des.patch
patching file src/openvpn/syshead.h
patching file src/openvpn/crypto_mbedtls.c

Now at patch ../patches/220-disable_des.patch
user@machine:/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_mips32_musl/openvpn-wolfssl/openvpn-2.5.2/patches$
```

```

**Comment by PolynomialDivision** [comment]
When I have time I will have a look if I find the error.

**Comment by mkrkn** [comment]
When you get back to this, please escape () in the package title with backslashes ``\`` instead of forward slashes.

The CI errors are likely caused by you copying the patch verbatim from OpenVPN. I remember I had the same issue backporting another patch - Quilt itself is fine with it, but the CI is more picky about formatting. Try setting up your .quiltrc as in [Working with patches](https://openwrt.org/docs/guide-developer/build-system/use-patches-with-buildsystem) and refreshing your patch.

**Comment by AuthorReflex** [comment]
> 
> 
> When you get back to this, please escape () in the package title with backslashes `\` instead of forward slashes.
> 
> The CI errors are likely caused by you copying the patch verbatim from OpenVPN. I remember I had the same issue backporting another patch - Quilt itself is fine with it, but the CI is more picky about formatting. Try setting up your .quiltrc as in [Working with patches](https://openwrt.org/docs/guide-developer/build-system/use-patches-with-buildsystem) and refreshing your patch.

Many thanks for your reply, fixed.

**Comment by AuthorReflex** [comment]
I reviewed the patch with ./scripts/checkpatch.pl and found one error. Error has been fixed

checkpatch.pl --no-tree 002-add-wolfssl-support.patch
total: 0 errors, 0 warnings, 139 lines checked

002-add-wolfssl-support.patch has no obvious style problems and is ready for submission.

**Comment by neheb** [comment]
You need to run

```
make package/openvpn/{clean,refresh}
```

**Comment by AuthorReflex** [comment]
> 
> 
> You need to run
> 
> ```
> make package/openvpn/{clean,refresh}
> ```

It's no problem
I found in patch some displacement in offset for reverse applying, fixed

**Comment by AuthorReflex** [comment]
> 
> 
> You need to run
> 
> ```
> make package/openvpn/{clean,refresh}
> ```

What validation is performed for patches and says 'dirty'?

**Comment by neheb** [comment]
> > You need to run
> > ```
> > make package/openvpn/{clean,refresh}
> > ```
> 
> What validation is performed for patches and says 'dirty'?

It runs through quilt. Don't bother trying to manually fix. Just run the command above.

**Comment by AuthorReflex** [comment]
> 
> 
> > > You need to run
> > > ```
> > > make package/openvpn/{clean,refresh}
> > > ```
> > 
> > 
> > What validation is performed for patches and says 'dirty'?
> 
> It runs through quilt. Don't bother trying to manually fix. Just run the command above.

No errors && messages

/mnt/D2/openwrt/openwrt-trunk make package/openvpn/{clean,refresh} V=s
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[1]: Entering directory '/mnt/D2/openwrt/openwrt-trunk'
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/feeds/packages/net/openvpn'
rm -rf /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2
rm -f /mnt/D2/openwrt/openwrt-trunk/staging_dir/target-mipsel_24kc_musl/stamp/.openvpn.wolfssl_installed
rm -f /mnt/D2/openwrt/openwrt-trunk/staging_dir/target-mipsel_24kc_musl/packages/openvpn.wolfssl.list
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/feeds/packages/net/openvpn'
time: package/feeds/packages/openvpn/wolfssl/clean#0.11#0.11#5.28
make[1]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk'
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[1]: Entering directory '/mnt/D2/openwrt/openwrt-trunk'
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/feeds/packages/net/openvpn'
mkdir -p /mnt/D2/openwrt/openwrt-trunk/dl
SHELL= flock /mnt/D2/openwrt/openwrt-trunk/tmp/.openvpn-2.5.2.tar.xz.flock -c '        /mnt/D2/openwrt/openwrt-trunk/scripts/download.pl "/mnt/D2/openwrt/openwrt-trunk/dl" "openvpn-2.5.2.tar.xz" "b12743836901f365efaf82ab2493967e1b21c21eb43ce9a8da1002a17c9c1dc8" "" "https://build.openvpn.net/downloads/releases/" "https://swupdate.openvpn.net/community/releases/"    '
touch /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.prepared_check
. /mnt/D2/openwrt/openwrt-trunk/include/shell.sh; xzcat /mnt/D2/openwrt/openwrt-trunk/dl/openvpn-2.5.2.tar.xz | tar -C /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.. -xf -
[ ! -d ./src/ ] || cp -fpR ./src/. /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2
rm -rf /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches; mkdir -p /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches
touch /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.quilt_used
touch /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.prepared
if [ -s "/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/series" ]; then (cd "/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2"; if quilt --quiltrc=- next >/dev/null 2>&1; then quilt --quiltrc=- push -a; else quilt --quiltrc=- top >/dev/null 2>&1; fi ); fi
Applying patch 001-reproducible-remove_DATE.patch
patching file src/openvpn/options.c

Applying patch 002-add-wolfssl-support.patch
patching file configure.ac
patching file src/openvpn/syshead.h

Applying patch 100-mbedtls-disable-runtime-version-check.patch
patching file src/openvpn/ssl_mbedtls.c

Applying patch 115-fix-mbedtls-without-renegotiation.patch
patching file src/openvpn/ssl_mbedtls.c

Applying patch 210-build_always_use_internal_lz4.patch
patching file configure.ac

Applying patch 220-disable_des.patch
patching file src/openvpn/syshead.h
patching file src/openvpn/crypto_mbedtls.c

Now at patch 220-disable_des.patch
touch "/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.quilt_checked"
Applying patch 001-reproducible-remove_DATE.patch
patching file src/openvpn/options.c

Now at patch 001-reproducible-remove_DATE.patch
Patch 001-reproducible-remove_DATE.patch is unchanged
Applying patch 002-add-wolfssl-support.patch
patching file configure.ac
patching file src/openvpn/syshead.h

Now at patch 002-add-wolfssl-support.patch
Refreshed patch 002-add-wolfssl-support.patch
Applying patch 100-mbedtls-disable-runtime-version-check.patch
patching file src/openvpn/ssl_mbedtls.c

Now at patch 100-mbedtls-disable-runtime-version-check.patch
Patch 100-mbedtls-disable-runtime-version-check.patch is unchanged
Applying patch 115-fix-mbedtls-without-renegotiation.patch
patching file src/openvpn/ssl_mbedtls.c

Now at patch 115-fix-mbedtls-without-renegotiation.patch
Patch 115-fix-mbedtls-without-renegotiation.patch is unchanged
Applying patch 210-build_always_use_internal_lz4.patch
patching file configure.ac

Now at patch 210-build_always_use_internal_lz4.patch
Patch 210-build_always_use_internal_lz4.patch is unchanged
Applying patch 220-disable_des.patch
patching file src/openvpn/syshead.h
patching file src/openvpn/crypto_mbedtls.c

Now at patch 220-disable_des.patch
Patch 220-disable_des.patch is unchanged
mkdir -p ./patches
rm -f ./patches/* 2>/dev/null >/dev/null
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/001-reproducible-remove_DATE.patch' -> './patches/001-reproducible-remove_DATE.patch'
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/002-add-wolfssl-support.patch' -> './patches/002-add-wolfssl-support.patch'
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/100-mbedtls-disable-runtime-version-check.patch' -> './patches/100-mbedtls-disable-runtime-version-check.patch'
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/115-fix-mbedtls-without-renegotiation.patch' -> './patches/115-fix-mbedtls-without-renegotiation.patch'
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/210-build_always_use_internal_lz4.patch' -> './patches/210-build_always_use_internal_lz4.patch'
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/220-disable_des.patch' -> './patches/220-disable_des.patch'
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/feeds/packages/net/openvpn'
time: package/feeds/packages/openvpn/wolfssl/refresh#1.17#0.64#2.81
make[1]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk'

**Comment by neheb** [comment]
```

Now at patch 220-disable_des.patch
touch "/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.quilt_checked"
Applying patch 001-reproducible-remove_DATE.patch
patching file src/openvpn/options.c
```

is the error.

**Comment by AuthorReflex** [comment]
> 
> 
> ```
> 
> Now at patch 220-disable_des.patch
> touch "/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.quilt_checked"
> Applying patch 001-reproducible-remove_DATE.patch
> patching file src/openvpn/options.c
> ```
> 
> is the error.

Fixed. All patches has been unchanged after refresh.
```
 make package/openvpn/{clean,refresh} V=sc
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[1]: Entering directory '/mnt/D2/openwrt/openwrt-trunk'
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/feeds/packages/net/openvpn'
rm -rf /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2
rm -f /mnt/D2/openwrt/openwrt-trunk/staging_dir/target-mipsel_24kc_musl/stamp/.openvpn.wolfssl_installed
rm -f /mnt/D2/openwrt/openwrt-trunk/staging_dir/target-mipsel_24kc_musl/packages/openvpn.wolfssl.list
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/feeds/packages/net/openvpn'
time: package/feeds/packages/openvpn/wolfssl/clean#0.08#0.05#0.28
make[1]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk'
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/scripts/config'
make[1]: Entering directory '/mnt/D2/openwrt/openwrt-trunk'
make[2]: Entering directory '/mnt/D2/openwrt/openwrt-trunk/feeds/packages/net/openvpn'
mkdir -p /mnt/D2/openwrt/openwrt-trunk/dl
SHELL= flock /mnt/D2/openwrt/openwrt-trunk/tmp/.openvpn-2.5.2.tar.xz.flock -c '        /mnt/D2/openwrt/openwrt-trunk/scripts/download.pl "/mnt/D2/openwrt/openwrt-trunk/dl" "openvpn-2.5.2.tar.xz" "b12743836901f365efaf82ab2493967e1b21c21eb43ce9a8da1002a17c9c1dc8" "" "https://build.openvpn.net/downloads/releases/" "https://swupdate.openvpn.net/community/releases/"    '
touch /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.prepared_check
. /mnt/D2/openwrt/openwrt-trunk/include/shell.sh; xzcat /mnt/D2/openwrt/openwrt-trunk/dl/openvpn-2.5.2.tar.xz | tar -C /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.. -xf -
[ ! -d ./src/ ] || cp -fpR ./src/. /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2
rm -rf /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches; mkdir -p /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches
touch /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.quilt_used
touch /mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.prepared
if [ -s "/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/series" ]; then (cd "/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2"; if quilt --quiltrc=- next >/dev/null 2>&1; then quilt --quiltrc=- push -a; else quilt --quiltrc=- top >/dev/null 2>&1; fi ); fi
Applying patch 001-reproducible-remove_DATE.patch
patching file src/openvpn/options.c

Applying patch 002-add-wolfssl-support.patch
patching file configure.ac
patching file src/openvpn/syshead.h

Applying patch 100-mbedtls-disable-runtime-version-check.patch
patching file src/openvpn/ssl_mbedtls.c

Applying patch 115-fix-mbedtls-without-renegotiation.patch
patching file src/openvpn/ssl_mbedtls.c

Applying patch 210-build_always_use_internal_lz4.patch
patching file configure.ac

Applying patch 220-disable_des.patch
patching file src/openvpn/syshead.h
patching file src/openvpn/crypto_mbedtls.c

Now at patch 220-disable_des.patch
touch "/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/.quilt_checked"
Applying patch 001-reproducible-remove_DATE.patch
patching file src/openvpn/options.c

Now at patch 001-reproducible-remove_DATE.patch
Patch 001-reproducible-remove_DATE.patch is unchanged
Applying patch 002-add-wolfssl-support.patch
patching file configure.ac
patching file src/openvpn/syshead.h

Now at patch 002-add-wolfssl-support.patch
Patch 002-add-wolfssl-support.patch is unchanged
Applying patch 100-mbedtls-disable-runtime-version-check.patch
patching file src/openvpn/ssl_mbedtls.c

Now at patch 100-mbedtls-disable-runtime-version-check.patch
Patch 100-mbedtls-disable-runtime-version-check.patch is unchanged
Applying patch 115-fix-mbedtls-without-renegotiation.patch
patching file src/openvpn/ssl_mbedtls.c

Now at patch 115-fix-mbedtls-without-renegotiation.patch
Patch 115-fix-mbedtls-without-renegotiation.patch is unchanged
Applying patch 210-build_always_use_internal_lz4.patch
patching file configure.ac

Now at patch 210-build_always_use_internal_lz4.patch
Patch 210-build_always_use_internal_lz4.patch is unchanged
Applying patch 220-disable_des.patch
patching file src/openvpn/syshead.h
patching file src/openvpn/crypto_mbedtls.c

Now at patch 220-disable_des.patch
Patch 220-disable_des.patch is unchanged
mkdir -p ./patches
rm -f ./patches/* 2>/dev/null >/dev/null
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/001-reproducible-remove_DATE.patch' -> './patches/001-reproducible-remove_DATE.patch'
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/002-add-wolfssl-support.patch' -> './patches/002-add-wolfssl-support.patch'
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/100-mbedtls-disable-runtime-version-check.patch' -> './patches/100-mbedtls-disable-runtime-version-check.patch'
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/115-fix-mbedtls-without-renegotiation.patch' -> './patches/115-fix-mbedtls-without-renegotiation.patch'
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/210-build_always_use_internal_lz4.patch' -> './patches/210-build_always_use_internal_lz4.patch'
'/mnt/D2/openwrt/openwrt-trunk/build_dir/target-mipsel_24kc_musl/openvpn-wolfssl/openvpn-2.5.2/patches/220-disable_des.patch' -> './patches/220-disable_des.patch'
make[2]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk/feeds/packages/net/openvpn'
time: package/feeds/packages/openvpn/wolfssl/refresh#1.16#0.71#1.67
make[1]: Leaving directory '/mnt/D2/openwrt/openwrt-trunk'
```

**Comment by neheb** [comment]
ping @mkrkn  <-- good to merge?

**Comment by mkrkn** [comment]
I'm OK with merging at this point.
